x/bridge/observer/bitcoin/bitcoin_test.go

```diff
@@ -63,110 +72,73 @@ func success(t *testing.T) http.HandlerFunc {
 	}
 }
 
-// TestObserverSuccess verifies Observer properly processes observed transactions
-func TestObserverSuccess(t *testing.T) {
-	s := httptest.NewServer(http.HandlerFunc(success(t)))
+func TestListenOutboundTransfer(t *testing.T) {
+	s := httptest.NewServer(success(t))
 	defer s.Close()
 
 	host, _ := strings.CutPrefix(s.URL, "http://")
-	cfg := bitcoin.RpcConfig{
-		Host:       host,
-		DisableTls: true,
-		User:       "test",
-		Pass:       "test",
-	}
+	client, err := rpcclient.New(&rpcclient.ConnConfig{
+		Host:         host,
+		DisableTLS:   true,
+		HTTPPostMode: true,
+		User:         "test",
+		Pass:         "test",
+		Params:       chaincfg.TestNet3Params.Name,
+	}, nil)
+	require.NoError(t, err)
 
 	initialHeight := uint64(2582657)
-	observer, err := bitcoin.NewObserver(
+	b, err := bitcoin.NewChainClient(
 		log.NewNopLogger(),
-		cfg,
-		"2N4qEFwruq3zznQs78twskBrNTc6kpq87j1",
-		initialHeight,
+		client,
+		BtcVault,
 		time.Second,
+		initialHeight,
+		chaincfg.TestNet3Params,
 	)
 	require.NoError(t, err)
 
-	observer.Start()
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	err = b.Start(ctx)
+	require.NoError(t, err)
 
 	// We expect Observer to observe 1 block with 2 Txs
 	// Only 1 Tx is sent to our vault address,
 	// so we should receive only 1 TxIn
-	txs := observer.TxIns()
-	var tx bitcoin.TxIn
+	txs := b.ListenOutboundTransfer()
+	var out observer.Transfer
 	require.Eventually(t, func() bool {
-		tx = <-txs
+		out = <-txs
 		return true
-	}, time.Second, 100*time.Millisecond, "Timeout reading events from observer")
-
-	expectedTx := bitcoin.TxIn{
-		Id:          "f395b2cc8551aff25fe8d61fec159a6b93d29b9ff56a68c9d29df99a864fd74c",
-		Height:      initialHeight,
-		Sender:      "2Mt1ttL5yffdfCGxpfxmceNE4CRUcAsBbgQ",
-		Destination: "2N4qEFwruq3zznQs78twskBrNTc6kpq87j1",
-		Amount:      math.NewUint(10000),
-		Memo:        "osmo13g23crzfp99xg28nh0j4em4nsqnaur02nek2wt",
+	}, time.Second, 100*time.Millisecond, "Timeout reading transfer")
+
+	expOut := observer.Transfer{
+		SrcChain: observer.ChainIdBitcoin,
+		DstChain: observer.ChainIdOsmosis,
+		Id:       "ef4cd511c64834bde624000b94110c9f184388566a97d68d355339294a72dadf",
+		Height:   initialHeight,
+		Sender:   "", // the sender is set in the osmosis chain client
+		To:       "osmo13g23crzfp99xg28nh0j4em4nsqnaur02nek2wt",
+		Asset:    string(bridgetypes.DefaultBitcoinDenomName),
+		Amount:   math.NewUint(10000),
```

Comment:
Do we test somewhere down the line if the observer cannot be set with invalid params? (i.e., we set the sender to a BTC address earlier, do we check/test that this is not possible?)